drivers/dht/dht.c

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2015 Ludwig Ortmann
+ * Copyright 2015 Christian Mehlis
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     driver_dht
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the DHT 11 and 22
+ *              temperature and humidity sensor
+ *
+ * @author      Ludwig Ortmann <ludwig.ortmann@fu-berlin.de>
+ * @author      Christian Mehlis <mehlis@inf.fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <stdint.h>
+
+#include "hwtimer.h"
+#include "timex.h"
+#include "periph/gpio.h"
+
+#include "dht.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/***********************************************************************
+ * internal API declaration
+ **********************************************************************/
+
+static void dht_read_data(gpio_t dev, uint32_t *data, uint8_t *checksum);
+static int dht_test_checksum(uint32_t data, uint8_t checksum);
+void dht_parse_11(dht_data_t *data, float *outhum, float *outtemp);
+void dht_parse_22(dht_data_t *data, float *outhum, float *outtemp);
+
+/***********************************************************************
+ * internal API implementation
+ **********************************************************************/
+
+void dht_parse_11(dht_data_t *data, float *outhum, float *outtemp)
+{
+    *outhum = data->humidity >> 8;
+    *outtemp = data->temperature >> 8;
+}
+
+void dht_parse_22(dht_data_t *data, float *outhum, float *outtemp)
+{
+    *outhum = data->humidity / 10;
```

Comment:
I didn't find this in the datasheet immediately but I guess this has to do with the higher resoultion of the dht22?